apps/src/code-studio/pd/form_components_func/FormController.jsx

```diff
@@ -328,41 +358,25 @@ const FormController = props => {
     setGlobalError(false);
     setSubmitting(true);
 
-    // [MEG] TODO: only do a post if it's a new application
-    $.ajax({
-      method: 'POST',
-      url: apiEndpoint,
-      contentType: 'application/json',
-      dataType: 'json',
-      data: JSON.stringify(serializeFormData())
-    })
-      .done(data => {
-        sessionStorage.removeItem(sessionStorageKey);
-        onSuccessfulSubmit(data);
-      })
-      .fail(data => {
-        if (
-          data.responseJSON &&
-          data.responseJSON.errors &&
-          data.responseJSON.errors.form_data
-        ) {
-          if (data.responseJSON.general_error) {
-            setErrors(data.responseJSON.errors.form_data);
-            setErrorHeader(data.responseJSON.general_error);
-            setGlobalError(true);
-          } else {
-            // if the failure was a result of an invalid form, highlight the errors
-            // and display the generic error header
-            setErrors(data.responseJSON.errors.form_data);
-            setErrorHeader(i18n.formErrorsBelow());
-          }
-        } else {
-          // Otherwise, something unknown went wrong on the server
-          setGlobalError(true);
-          setErrorHeader(i18n.formServerError());
-        }
-        setSubmitting(false);
-      });
+    const handleSuccessfulSubmit = data => {
+      sessionStorage.removeItem(sessionStorageKey);
+      onSuccessfulSubmit(data);
+    };
+
+    const handleRequestFailure = data => {
```

Comment:
naming nit: is this going to be called for any request failure ( save or submit), or only on submit? I like that you named `handleSuccessfulSubmit` specifically, since it makes it easier to read knowing that method is specific to submit. if this one is specific to submit, it would be nice to make this method name specific to submit as well.